test/Transaction.test.js

```diff
@@ -186,6 +187,55 @@ describe("Transaction", () => {
         const onceConfirmedReceiptSpy = sinon.spy();
         const onceTxRevertSpy = sinon.spy();
 
+        let errorCaught;
+        try {
+            tx.send({ from: "0x0" })
+                .onceTxHash(onceTxHashSpy)
+                .onceReceipt(onceReceiptSpy)
+                .onConfirmation(onConfirmationSpy)
+                .onceConfirmedReceipt(3, onceConfirmedReceiptSpy)
+                .onceTxRevert(onceTxRevertSpy);
+        } catch (error) {
```

Comment:
There should be a `fail` within the try block to ensure an exception was thrown. Same pattern should apply to the subsequent checks (instead of `isUndefined`).

Reply:
What you mean by fail within the block? `.send` should throw, that's what testing here. As for txHash `isUndefined` I'm not sure what you mean. Am I miss something? :)

Reply:
The pattern to properly check exceptions:

```
try {
  doStuff();
  assert.fail();   // we shouldn’t reach this line
} catch (e) {
  // validate the error
}
```

Reply:
Btw, you could also use `fail` for callbacks that shouldn’t be called. Maybe nicer than spy...